lib/media/streaming_engine.js

```diff
@@ -1708,6 +1732,54 @@ shaka.media.StreamingEngine = class {
     }
   }
 
+  /**
+   * Parse PRFT box.
+   * @param {!shaka.media.SegmentReference} reference
+   * @param {!shaka.extern.ParsedBox} box
+   * @private
+   */
+  parsePrft_(reference, box) {
+    if (this.parsedPrftEventRaised_) {
+      return;
+    }
+    box.reader.readUint32(); // Ignore referenceTrackId
+    const ntpTimestampSec = box.reader.readUint32();
+    const ntpTimestampFrac = box.reader.readUint32();
+    const ntpTimestamp = ntpTimestampSec * 1000 + ntpTimestampFrac / 2**32;
+
+    let mediaTime;
+    if ( box.version === 0) {
+      mediaTime = box.reader.readUint32();
+    } else {
+      mediaTime = box.reader.readUint64();
+    }
+    const wallClockTime = this.convertNtp(ntpTimestamp);
+    const prftInfo = {
+      wallClockTime,
+      mediaTime,
+      timescale: reference.initSegmentReference.timescale,
```

Comment:
Most parts of the Shaka API trade in seconds, instead.  Would it be enough to surface the presentation time instead by combining mediaTime and timescale?

Reply:
Do you mean just to surface the program start date i.e. `new Date(event.wallClockTime - (event.mediaTime / event.timescale) * 1000)`, and not surface timescale, mediatime, etc...?

Reply:
That could work.

I was actually suggesting just putting the media time figure into seconds and calling it a presentation time.  As is, you have mediaTime in timescale units, and wall clock time in milliseconds.  We normally trade in seconds for presentation times and Date for wall-clock times.

But if you want to combine it all into a start time as a Date, that's fine, too.

Reply:
@joeyparrish Great idea! I have made and tested those changes.